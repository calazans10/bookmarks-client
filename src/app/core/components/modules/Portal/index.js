import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';

class Portal extends React.Component {
  constructor(props) {
    super(props);

    this.modalRoot = document.getElementById('modal-root');
    this.el = document.createElement('div');
  }

  componentDidMount() {
    this.modalRoot.appendChild(this.el);
  }

  componentWillUnmount() {
    this.modalRoot.removeChild(this.el);
  }

  render() {
    return ReactDOM.createPortal(this.props.children, this.el);
  }
}

Portal.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Portal;
