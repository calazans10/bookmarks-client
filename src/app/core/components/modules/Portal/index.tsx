import React from 'react';
import ReactDOM from 'react-dom';

class Portal extends React.Component {
  modalRoot: HTMLElement
  el: HTMLDivElement


  constructor(props) {
    super(props);

    this.modalRoot = document.getElementById('modal-root')! as HTMLElement;
    this.el = document.createElement('div');
  }

  componentDidMount() {
    this.modalRoot.appendChild(this.el);
  }

  componentWillUnmount() {
    this.modalRoot.removeChild(this.el);
  }

  render() {
    return ReactDOM.createPortal(this.props.children, this.el);
  }
}

export default Portal;
