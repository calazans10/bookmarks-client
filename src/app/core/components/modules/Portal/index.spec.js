import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Portal from './index';

describe('Portal', () => {
  let modalRoot = document.getElementById('modal-root');
  if (!modalRoot) {
    modalRoot = document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    document.body.appendChild(modalRoot);
  }

  afterEach(cleanup);

  it('should render the Portal component', () => {
    const { container } = render(
      <Portal>
        <div>test</div>
      </Portal>
    );
    expect(container).toMatchSnapshot();
  });
});
