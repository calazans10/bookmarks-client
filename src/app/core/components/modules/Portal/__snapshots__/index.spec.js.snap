// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal should render the Portal component 1`] = `<div />`;
